tests/array_interoperability_test.py

```diff
@@ -274,6 +274,85 @@ def testJaxToCuPy(self, shape, dtype):
                      z.__cuda_array_interface__["data"][0])
     self.assertAllClose(x, cupy.asnumpy(z))
 
+  @unittest.skipIf(xla_extension_version < 237, "Requires newer jaxlib")
+  @jtu.sample_product(
+    shape=all_shapes,
+    dtype=jtu.dtypes.supported(cuda_array_interface_dtypes),
+  )
+  @unittest.skipIf(not cupy, "Test requires CuPy")
+  @jtu.run_on_devices("cuda")
+  def testCuPyToJax(self, shape, dtype):
+    rng = jtu.rand_default(self.rng())
+    x = rng(shape, dtype)
+    y = cupy.asarray(x)
+    z = jnp.array(y, copy=False)  # this conversion uses dlpack protocol
+    self.assertEqual(z.dtype, dtype)
+    self.assertEqual(y.__cuda_array_interface__["data"][0],
+                     z.__cuda_array_interface__["data"][0])
+    self.assertAllClose(np.asarray(z), cupy.asnumpy(y))
+
+  @unittest.skipIf(xla_extension_version < 237, "Requires newer jaxlib")
+  @jtu.sample_product(
+    shape=all_shapes,
+    dtype=jtu.dtypes.supported(cuda_array_interface_dtypes),
+  )
+  @jtu.run_on_devices("cuda")
+  def testCaiToJax(self, shape, dtype):
```

Comment:
We need a skip condition here for PJRT runtimes, similar to others in this file. Something like this should work:
```suggestion
  def testCaiToJax(self, shape, dtype):
    if xb.using_pjrt_c_api():
      self.skipTest("cuda_array_interface support is incomplete in the PJRT C API")  # TODO(jakevdp)
```

Reply:
Applied.

IIUC, PJRT is about running code on Intel GPUs but this test is decorated with `jtu.run_on_devices("cuda")` that would imply that the test ought to be skipped anyway when using a non-CUDA device. Is there an explanation why this test is still executed for PJRT runtimes?

Another question is if CAI support can ever be used within PJRT because CUDA is Nvidia device-specific?